superset-frontend/src/dashboard/components/SliceHeader/SliceHeader.test.tsx

```diff
@@ -0,0 +1,399 @@
+  const props = createProps();
+
+  // @ts-ignore
+  delete props.forceRefresh;
+  // @ts-ignore
+  delete props.updateSliceName;
+  // @ts-ignore
+  delete props.toggleExpandSlice;
+  // @ts-ignore
+  delete props.exploreChart;
+  // @ts-ignore
+  delete props.exportCSV;
+  // @ts-ignore
+  delete props.innerRef;
+  // @ts-ignore
+  delete props.editMode;
+  // @ts-ignore
+  delete props.annotationQuery;
+  // @ts-ignore
+  delete props.annotationError;
+  // @ts-ignore
+  delete props.cachedDttm;
+  // @ts-ignore
+  delete props.updatedDttm;
+  // @ts-ignore
+  delete props.isCached;
+  // @ts-ignore
+  delete props.isExpanded;
+  // @ts-ignore
+  delete props.sliceName;
+  // @ts-ignore
+  delete props.supersetCanExplore;
+  // @ts-ignore
+  delete props.supersetCanCSV;
+  // @ts-ignore
+  delete props.sliceCanEdit;
+
+  render(<SliceHeader {...props} />, { useRedux: true });
+  userEvent.click(screen.getByTestId('toggleExpandSlice'));
+  userEvent.click(screen.getByTestId('forceRefresh'));
+  userEvent.click(screen.getByTestId('exploreChart'));
+  userEvent.click(screen.getByTestId('exportCSV'));
+  userEvent.click(screen.getByTestId('addSuccessToast'));
+  userEvent.click(screen.getByTestId('addDangerToast'));
+  userEvent.click(screen.getByTestId('handleToggleFullSize'));
+  expect(screen.getByTestId('slice-header')).toBeInTheDocument();
+});
+
+test('Should render title', () => {
+  const props = createProps();
+  render(<SliceHeader {...props} />, { useRedux: true });
+  expect(screen.getByText('Vaccine Candidates per Phase')).toBeInTheDocument();
+});
+
+test('Should render "annoationsLoading"', () => {
+  const props = createProps();
+  render(<SliceHeader {...props} />, { useRedux: true });
+  expect(
+    screen.getByRole('img', {
+      name: 'Annotation layers are still loading.',
+    }),
+  ).toBeInTheDocument();
+});
+
+test('Should render "annoationsError"', () => {
+  const props = createProps();
+  render(<SliceHeader {...props} />, { useRedux: true });
+  expect(
+    screen.getByRole('img', {
+      name: 'One ore more annotation layers failed loading.',
+    }),
+  ).toBeInTheDocument();
+});
+
+test('Should not render "annoationsError" and "annoationsLoading"', () => {
+  const props = createProps();
+  props.annotationQuery = {};
+  props.annotationError = {};
+  render(<SliceHeader {...props} />, { useRedux: true });
+  expect(
+    screen.queryByRole('img', {
+      name: 'One ore more annotation layers failed loading.',
+    }),
+  ).not.toBeInTheDocument();
+  expect(
+    screen.queryByRole('img', {
+      name: 'Annotation layers are still loading.',
+    }),
+  ).not.toBeInTheDocument();
+});
+
+test('Correct props to "FiltersBadge"', () => {
+  const props = createProps();
+  render(<SliceHeader {...props} />, { useRedux: true });
+  expect(screen.getByTestId('FiltersBadge')).toHaveAttribute(
+    'data-chart-id',
+    '312',
+  );
+});
+
+test('Correct props to "SliceHeaderControls"', () => {
```

Comment:
I think that `SliceHeader` and `SliceHeaderControls` are the same feature, they should be in the same folder. If they are grouped we could have tests that consider this composition and we can discard all this mocking code. This would make the tests more similar to what the user is actually going to experience. Let's make a call so I can explain this a little better.

Reply:
@yardz did you intend to make any changes to this PR based on this discussion?

Reply:
@rusackas I had already talked to @michael-s-molina but forgot to put the answer here. hahaha

I agree with him that they should be in the same folder. But as I am opening up component by component I would not be able to do this in a single PR.

And the IAPA had also not been approved. I think these two components should be put together when we are rearranging the files.